src/component/routers/WebhookRouter.tsx

```diff
@@ -72,39 +52,35 @@ export const mapHeaders = (headers: Headers): Header[] =>
         uuid: generateUUID()
     }));
 
-export const getInitialState = (action: CallWebhook): WebhookState => {
+export const getInitialState = (action: CallWebhook): WebhookRouterState => {
+    let state = initialState;
     if (action.type === 'call_webhook') {
-        initialState.method = action.method;
+        state = { ...state, method: action.method };
         if (action.headers && Object.keys(action.headers).length) {
             const existingHeaders = mapHeaders(action.headers);
-            initialState.headers.unshift(...existingHeaders);
+            state = {
```

Comment:
is this OK? just trying to keep state pure as we have render two `WebhookRouter` components when the user switches to that form

Reply:
ya, I think this is alright.. at least the object being spread is right there above it so its pretty clear what's going on there.